saleor/tests/e2e/checkout/test_checkout_purchase_with_no_payment.py

```diff
@@ -0,0 +1,173 @@
+    )
+    channel_id = channel_data["id"]
+
+    channel_ids = [channel_id]
+    shipping_zone_data = create_shipping_zone(
+        e2e_staff_api_client,
+        warehouse_ids=warehouse_ids,
+        channel_ids=channel_ids,
+    )
+    shipping_zone_id = shipping_zone_data["id"]
+
+    shipping_method_data = create_shipping_method(
+        e2e_staff_api_client, shipping_zone_id
+    )
+    shipping_method_id = shipping_method_data["id"]
+
+    create_shipping_method_channel_listing(
+        e2e_staff_api_client, shipping_method_id, channel_id
+    )
+
+    product_type_data = create_product_type(
+        e2e_staff_api_client,
+    )
+    product_type_id = product_type_data["id"]
+
+    category_data = create_category(e2e_staff_api_client)
+    category_id = category_data["id"]
+
+    product_data = create_product(e2e_staff_api_client, product_type_id, category_id)
+    product_id = product_data["id"]
+
+    create_product_channel_listing(e2e_staff_api_client, product_id, channel_id)
+
+    stocks = [
+        {
+            "warehouse": warehouse_id,
+            "quantity": 5,
+        }
+    ]
+    product_variant_data = create_product_variant(
+        e2e_staff_api_client,
+        product_id,
+        stocks=stocks,
+    )
+    product_variant_id = product_variant_data["id"]
+
+    create_product_variant_channel_listing(
+        e2e_staff_api_client,
+        product_variant_id,
+        channel_id,
+    )
+
+    return product_variant_id
+
+
+@pytest.mark.e2e
+def test_should_not_be_able_to_make_purchase_with_no_payment(
+    e2e_staff_api_client,
+    e2e_not_logged_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_shipping,
+    permission_manage_product_types_and_attributes,
+):
+    # Before
+    channel_slug = "test-channel"
+    product_variant_id = prepare_product(
+        e2e_staff_api_client,
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        channel_slug,
+    )
+
+    # Step 1 - Create checkout.
+    lines = [
+        {"variantId": product_variant_id, "quantity": 1},
+    ]
+    checkout_data = checkout_create(
+        e2e_not_logged_api_client,
+        lines,
+        channel_slug,
+        email="testEmail@example.com",
+        set_default_billing_address=True,
+    )
+    checkout_id = checkout_data["id"]
+
+    assert checkout_data["isShippingRequired"] is True
+    assert checkout_data["shippingMethods"] == []
+    assert checkout_data["deliveryMethod"] is None
+    assert checkout_data["shippingMethod"] is None
+
+    # Step 2 - Set shipping address and DeliveryMethod for checkout
+    checkout_data = checkout_shipping_address_update(
+        e2e_not_logged_api_client, checkout_id
+    )
+    assert len(checkout_data["shippingMethods"]) == 1
+    shipping_method_id = checkout_data["shippingMethods"][0]["id"]
```

Comment:
Maybe we should set the shipping address in the checkout creation? 
Due to that, we perform fewer requests in this test. 
```suggestion
    checkout_data = checkout_create(
        e2e_not_logged_api_client,
        lines,
        channel_slug,
        email="testEmail@example.com",
        set_default_billing_address=True,
        set_default_shipping_address=True,
    )
    checkout_id = checkout_data["id"]

    assert checkout_data["isShippingRequired"] is True
    assert checkout_data["deliveryMethod"] is None
    assert checkout_data["shippingMethod"] is None
    shipping_method_id = checkout_data["shippingMethods"][0]["id"]

    # Step 2 - Set shipping address and DeliveryMethod for checkout    
```